packages/epo-react-lib/src/form/SelectListbox/SelectListbox.tsx

```diff
@@ -0,0 +1,141 @@
+import { FunctionComponent, useLayoutEffect, useState } from "react";
+import { useUID } from "react-uid";
+import { useTranslation } from "react-i18next";
+import { onChangeCallback, Option } from "@/types/select-listbox";
+import IconComposer from "@/svg/IconComposer";
+import * as Styled from "./styles";
+import { useAccessibleDropdown, useOnOutsideClick } from "./hooks";
+
+interface SelectListboxProps {
+  value: string | string[] | null;
+  options: Option[];
+  onChangeCallback: onChangeCallback;
+  isDisabled?: boolean;
+  placeholder?: string;
+  labelledById?: string;
+  namespace?: string;
+  isMultiselect?: boolean;
+  maxWidth?: string;
+  width?: string;
+}
+
+const SelectListbox: FunctionComponent<SelectListboxProps> = ({
+  value = null,
+  options,
+  onChangeCallback,
+  isDisabled,
+  labelledById,
+  placeholder: unsafePlaceholder,
+  namespace,
+  isMultiselect = false,
+  maxWidth = "200px",
+  width,
+}) => {
+  const uid = namespace || useUID();
+  const {
+    isDropdownOpen,
+    setIsDropdownOpen,
+    activeIndex,
+    setActiveIndex,
+    select,
+    setIsFocus,
+    listRef,
+    buttonRef,
+  } = useAccessibleDropdown({
+    options,
+    value,
+    onChangeCallback,
+    isMultiselect,
+  });
+  const [selectWidth, setSelectWidth] = useState(0);
+  const { t } = useTranslation();
+
+  useLayoutEffect(() => {
```

Comment:
to size the sibling button based on the list dropdown we need to get the size of the dropdown, unfortunately the dropdown itself is absolutely positioned so we can't use CSS since the dropdown will not influence the size of the parent. `useLayoutEffect()` gets the size of the dropdown ref at the earliest possible point, this still isn't perfect thought since there is a very brief resize flash.